comp/rdnsquerier/mock/mock.go

```diff
@@ -51,3 +53,18 @@ func (q *rdnsQuerierMock) GetHostname(ipAddr []byte, updateHostnameSync func(str
 
 	return nil
 }
+
+// GetHostnameSync simulates resolving the hostname for the given IP address synchronously.  If the IP address is in the private address
+// space then the resolved hostname is returned.
+func (q *rdnsQuerierMock) GetHostnameSync(_ context.Context, ipAddr string) (string, error) {
+	ipaddr := net.ParseIP(ipAddr).To4()
```

Comment:
This should be able work for IPv6 similarly to the comment above in the real implementation.